components/mailchimp/mailchimp.app.js

```diff
@@ -0,0 +1,681 @@
+     * @param {Integer} config.offset - For pagination, this the number of records
+     * from a collection to skip. Default value is 0.
+     * @param {String} config.sort_field - Returns files sorted by the specified
+     * field. Possible values: "created_at", "updated_at", or "end_time".
+     * @param {String} config.sort_dir - Determines the order direction for
+     * sorted results. Possible values: "ASC" or "DESC".
+     * @returns {facebook_ads: array, total_items: integer, _links: object }
+     * An array with the information of the `facebook_ads` returned, `total_items`
+     * with the total count of the facebook_ads (the outreach) collection,
+     * and `_links` a list of link types and descriptions for the API schema documents.
+     */
+    async getAllFacebookAds(config) {
+      const mailchimp = this.api();
+      return await this._withRetries(() =>
+        mailchimp.facebookAds.list(config));
+    },
+    /**
+     * Gets files in the File Manager under the connected Mailchimp acccount.
+     * @param {Object} config - An object representing the configuration options for this method.
+    * @param {Integer} config.count - For pagination, the number of records to return on each page.
+    * Default value is 10. Maximum value is 1000.
+     * @param {Integer} config.offset - For pagination, this the number of records from a collection
+     * to skip. Default value is 0.ssaa
+     * @param {String} config.type - The file type for the File Manager file, can be "image" or
+     * "file". Don't specify for both types.
+     * @param {Boolean} config.includeTransactional - Include transactional members in response.
+     * @param {Date} config.beforeCreatedAt - Restrict response to files created before the set
+     * date. Uses ISO 8601 time format: 2015-10-21T15:41:36+00:00.
+     * @param {Date} config.sinceCreatedAt - Restrict response to files created since the set date.
+     * Uses ISO 8601 time format: 2015-10-21T15:41:36+00:00.
+     * @returns {files: array, total_file_size: integer, total_items: integer, _links: object } An
+     * array with the information of the `files` returned, `total_file_size` with the total size of
+     * all File Manager files in bytes, `total_items` with the total count of the files
+     * collection, and `_links` array of links for file manipulation through the Mailchimp API.
+     */
+    async *getFileStream(fileType = "all", config) {
+      let offset = 0;
+      const type = fileType === "all"
+        ? null
+        : fileType;
+        const opts = {
+          offset,
+          type,
+          ...config
+        };
+        do {
+          opts.offset = offset;
+          const { files = [] } = await this._withRetries(
+            () => this.api().fileManager.files(opts),
+          );
+          if (files.length === 0) {
+            return;
+          }
+          for (const file of files) {
+            yield file;
+          }
+        offset += files.length;
```

Comment:
The indentation looks a bit off here.
```suggestion
      const type = fileType === "all"
        ? null
        : fileType;
      const opts = {
        offset,
        type,
        ...config
      };
      do {
        opts.offset = offset;
        const { files = [] } = await this._withRetries(
          () => this.api().fileManager.files(opts),
        );
        if (files.length === 0) {
          return;
        }
        for (const file of files) {
          yield file;
        }
        offset += files.length;
```